borg/helpers.py

```diff
@@ -1256,3 +1256,20 @@ def format_time(self, key, item):
 
     def time(self, key, item):
         return safe_timestamp(item.get(key) or item[b'mtime'])
+
+
+COMPRESS_NONE = set("""
+.gz .tgz .bz2 .tbz2 .xz .lzma .lzo .zip .rar .7z
+.mp3 .ogg .oga .flac
+.mp4 .mkv .avi .mpg .mpeg .webm .vob .ts .ogv
+.jpg .jpeg .png .gif
```

Comment:
Optional file for compression schemes:
```
lzma,9:<pattern>
zlib,3:<pattern>
```
With the usual support for patterns, which would then easily allow **both** to add extensions, and to add specific paths, with semantics that we already use elsewhere.

The syntax for this compression mode would then be, e.g. `--compression auto,[<number>,]<fallback>`, with `<fallback>` being any normal compression method, that is used on files that don't match any defined pattern.

We can then use LZ4 on a chunk-per-chunk basis to decide whether LZ4 or the `<fallback>` is used. (As I argued below, LZ4 is extremely fast and running it over all chunks that should receive a stronger, much slower compression algorithm will make no tangible difference. Some Googling in the mean time also turned up that LZ4 and some other algorithms are specifically enhanced to be very fast for incompressible inputs).

The optional `<number>` is the threshold used on the LZ4 ratio (default e.g. 75 %).

**Q:** Where to store the file, well-known position (e.g. ~/.config/borg/auto-compression), or repo-dependent / user-supplied
**Q:** Threshold default value?